docs/content/about-iota/iota-architecture/transaction-lifecycle.mdx

```diff
@@ -17,127 +17,107 @@ import ThemedImage from '@theme/ThemedImage';
+submitTransaction(transaction);
+```
 
-### Transaction Finalization
+## 3. Validation and Inclusion
+Full nodes perform consensus checks:
 
-The full node broadcasts this transaction _certificate_ to all validators. Validators verify the _certificate_, execute the transaction, and unlock the previously locked objects. They then sign the transaction effects (a detailed list of changes) and return these signed effects to the full node. The full node verifies these effects and, once it has enough signatures, forms an effects _certificate_.
+- Ensuring the transaction references valid parent transactions.
+- Checking the sender has sufficient funds.
+- Validating the digital signature.
 
-Your wallet app receives this effects _certificate_, which you can show to the coffee shop to prove that the transaction is complete and irreversible.
+If valid, the transaction enters the **mempool** and is broadcast to other nodes.
 
-## Transaction Life Cycle
+## 4 Epoch Transitions and Their Impact
 
-### Submission
+The IOTA network undergoes [epoch](./epochs.mdx) changes, which affect transaction processing.
 
-If you want to transfer an NFT from your wallet to a friend's wallet, you would create a transaction using a wallet app. This transaction includes the gas payment and a command to transfer the NFT. Before sending this transaction to the network, the wallet app needs to sign it. Once signed, the wallet app **submits** the transaction to an IOTA full node.
+- **Epoch Boundaries & Transaction Resubmission**: If a transaction is submitted near an epoch boundary, it may need resubmission in the new epoch.
+- **Who handles resubmission?** Full nodes may need to resubmit transactions in the new epoch. If using a client or wallet app, check if automatic resubmission is implemented.
+- **Should users avoid transactions during epoch changes?** While not required, it may help to wait for the epoch transition to complete before submitting time-sensitive transactions.
 
-### Certification
+### Transaction Certificates and Epochs
 
-After submission to a full node, the certification process begins. The full node can't certify the transaction alone because it does not have the full view of the network's transactions. Therefore, it sends the transaction to a validator. The validator checks if the transaction is valid and signs it if it is. The checks include:
+- **Certificates from the previous epoch become invalid** after an epoch change.
+- **What is included in a certificate?** At minimum, the epoch number.
+- **How is verification performed?** Validators’ public keys for the relevant epoch must be fetched from the network.
 
-* A valid user signature.
-* The transaction's initiator owns the input objects (e.g., the NFT being transferred).
-* All shared input objects in the transaction exist.
-* The gas payment is sufficient.
+## 5. Certification and Finalization
```

Comment:
Also this looks like something is missing. So we know the tx entered the mempool. And now it is already confirmed by validators?

Reply:
And here we say finalised after the validators confirmed. But Vlad mentioned checkpoints. So I agree we need to better describe what each step means for the user.

Reply:
I reworked it a little but please give more suggestions on my update here.